### PR TITLE
Change Behviour of non-uniform-work-group tests for OpenCL-3.0

### PR DESCRIPTION
**Pull request by FranklandJack**
* Pass `-cl-std=CL3.0` as build option for OpenCL-3.0.
* For an OpenCL-3.0 driver that optionally does not support non-uniform
work-groups round up the global size and run the tests with uniform
work-groups. This increases coverage and allows testing of the
`get_enqueued_local_size` builtin for uniform work-groups.

### CODE REVIEW

File: test_conformance/non_uniform_work_group/TestNonUniformWorkGroup.cpp
Comment (gwawiork) [inline test_conformance/non_uniform_work_group/TestNonUniformWorkGroup.cpp:542]: I am not sure if this is needed. 
@bashbaug do we need to change OpenCL C language version in this and other cases ?

Comment (bashbaug) [reply test_conformance/non_uniform_work_group/TestNonUniformWorkGroup.cpp:544]: Yes, I believe so.  I didn't do an exhaustive check, but I do see that the kernel string for this test includes the `get_enqueued_local_size()` built-in function, which requires OpenCL C 2.0 or OpenCL C 3.0 and was not in earlier versions of OpenCL C.

Comment (Jeremy-Kemp) [reply test_conformance/non_uniform_work_group/TestNonUniformWorkGroup.cpp:544]: With #879 , the selection could be completely omitted.

File: test_conformance/non_uniform_work_group/TestNonUniformWorkGroup.cpp
Comment (Jeremy-Kemp) [inline test_conformance/non_uniform_work_group/TestNonUniformWorkGroup.cpp:836]: With #879 , the selection could be completely omitted.

File: test_conformance/non_uniform_work_group/TestNonUniformWorkGroup.cpp
Comment (Jeremy-Kemp) [inline test_conformance/non_uniform_work_group/TestNonUniformWorkGroup.cpp:271]: I think this should be `>= Version(3, 0)` to ensure that future device versions remain compatible here.

Even if we were to make non-uniform a requirement in future versions of the spec, the `CL_DEVICE_NON_UNIFORM_WORK_GROUP_SUPPORT` query would still need to be supported and would therefore always return `CL_TRUE` in that case.

**Review by Jeremy-Kemp — CHANGES_REQUESTED** [review]

**Comment by FranklandJack** [comment]
Since #879 has been merged, this could do with another review before being merged. The clang format changes required to pass travis are quite drastic and obfuscate the functional changes of this patch, so I'll hold off on applying `clang-format` until this PR has been reviewed.

**Comment by FranklandJack** [comment]
I've clang formatted the patch, this hasn't introduced any functional changes, but the PR might need approving again.

**Comment by bashbaug** [comment]
Whoops, I didn't mean to close this PR.

I'll merge this as soon as the formatting issue is fixed.

**Comment by FranklandJack** [comment]
@bashbaug Looks like the formatting job has passed, needs approving one last time since I pushed new commits, but should be good to merge otherwise :)

**Pull request merged**
